Client/mods/deathmatch/logic/luadefs/CLuaDrawingDefs.cpp

```diff
@@ -566,6 +568,98 @@ int CLuaDrawingDefs::DxDrawImageSection(lua_State* luaVM)
     return 1;
 }
 
+int CLuaDrawingDefs::DxDrawPrimitive3D(lua_State* luaVM)
+{
+    // bool DxDrawPrimitive3D (string primitiveType, bool postGUI, table vertice1, ...)
+    D3DPRIMITIVETYPE    ePrimitiveType;
+    auto                pVecVertices = new std::vector<PrimitiveVertice>();
+    bool                bPostGUI;
+    CScriptArgReader argStream(luaVM);
+    argStream.ReadEnumString(ePrimitiveType);
+    argStream.ReadBool(bPostGUI, false);
+
+    while (argStream.NextIsTable())
+    {
+        std::vector<float> vecTableContent;
```

Comment:
This **std::vector** is going to heap allocate memory for every single vertice. Move it out of the **while** loop and call `std::vector<T>::clear()` here. Don't worry, `.size()` will have the expected value.

Reply:
@CrosRoad95 It seems `vecTableContent.clear()` is still missing in the latest version?

Reply:
@botder I assume these changes should be done for `DxDrawPrimitive` and `DxDrawMaterialPrimitive` as well, although a different issue?

Reply:
@CrosRoad95 
> and call std::vector<T>::clear() here
seems to be missing or reason why not?